File: codes/liskov/liskhely/liskh.java
class Bird
{
        
    public void  fly()
    {
            System.out.println("Am flying...\n");
    }
}

class Eagle extends  Bird
{
    public void fly()
        {
        System.out.println("Eagle: flying..\n");
        }
}
class Penguin extends  Bird
{
}

class Liskov
{

    public static void flyBird(Bird b)
    {
        b.fly();
    }

    public static void main(String[] args)
    {
        Bird theEagle = new Eagle();
        Bird thePenguin = new Penguin();
        
        flyBird(theEagle);
        flyBird(thePenguin);
        
        
    }

}

File: codes/liskov/liskhely/liskh.cpp
#include <iostream>

class Bird
{
    
public:
   
    virtual void  fly()
    {
        std::cout << "Am flying...\n" ;
    }
};

class Eagle : public Bird
{
public:
  
    void fly() override
    {
        std::cout << "Eagle: flying..\n";
    }
};

class Penguin : public Bird
{
};

static void flyBird(Bird& b)
{
    b.fly();
}

int main()
{
    Eagle theEagle;
    Penguin thePenguin;
    
    flyBird(theEagle);
    flyBird(thePenguin);
    
    return 0;
}

